### DISCUSSION TITLE
BulkWrite does not have a code to prevent unintentional operations.

### DISCUSSION THREAD

**Discussion by sunny4249** [BeanieODM/beanie#327] [General]
Hello, 
I have used beanie for simple interface easy to interact with collections. Also, It is good to make my code simple.

But while using bulk write context, I faced with a problem.
Below code does not have a code to prevent unintentional operation If  a code block in bulk write  context fail.
I think It is a problem, if raised, make it difficult to find bugs. 

It is okay?  

``` python
class BulkWriter:
    ...
    async def __aexit__(self, exc_type, exc, tb):
        await self.commit()

   ...
```

This is minimum code to prevent the problem. 

``` python
class BulkWriter:
    ...
    async def __aexit__(self, exc_type, exc, tb):
        if exec_type is not None:
            await self.commit()
   ...
```

**Comment by roman-right** [comment]
Hey! Thank you! I'll add it in one of the next releases
